stock_operation_package_mandatory/models/stock_picking.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+from odoo.tools import float_compare
+
+
+class StockPicking(models.Model):
+
+    _inherit = 'stock.picking'
+
+    @api.multi
+    def _check_mandatory_packages(self):
+        """ Check package existence for qty_done operations > 0
+        """
+        for picking in self:
+            if picking.picking_type_id.destination_package_mandatory:
+                operations = picking.pack_operation_product_ids.filtered(
+                    lambda o: float_compare(
+                        o.qty_done,
+                        0,
+                        o.product_id.uom_id.rounding) > 0)
+                if not all([operation.result_package_id
+                            for operation in operations]):
+                    raise UserError(
```

Comment:
@rousseldenis How can we provide to the user the informations required to solve the problem? It could be hard to find the operation where the package is missing...